text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+                "startColumn": { "type": "integer" },
+                "endLine": { "type": "integer" },
+                "endColumn": { "type": "integer" }
+            }
+        },
+        "StaticResourceReference": {
+            "type": "object",
+            "properties": {
+                "type": { "$ref": "#/definitions/StaticResourceType" },
+                "value": { 
+                    "anyOf": [
+                        { "type": "string", "format": "uri" },
+                        { "$ref": "#/definitions/ComputedValue"}
+                    ]
+                },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "StaticResourceType": {
+            "type": "string",
+            "enum": [ "image", "css", "html", "js", "other"]
+        },
+        "TemplateForEachDirective": {
+            "type": "string",
+            "properties": {
+                "items": { "$ref": "#/definitions/ComputedValue"  },
+                "itemName": { "type": "string"},
+                "indexName": { "type": "string" },
+                "key": {"type": "string" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "TemplateIfDirective": {
+            "type": "string",
+            "properties": {
+                "qualifier": {
+                    "type": "string",
+                    "enum": [ "true", "false" ]
+                },
+                "value": { "$ref": "#/definitions/ComputedValue"  }
+            }
+        },
+        "TemplateIteratorDirective": {
+            "type": "string",
+            "properties": {
+                "items": { "$ref": "#/definitions/ComputedValue"  },
+                "key": {"type": "string" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "TemplateLwcDomDirective": {
+            "type": "string",
+            "properties": {
+                "value": {
+                    "type": "string",
+                    "enum": [ "manual" ]
+                },
+                "tagName": { "type": "string" }
+            }
+        },
+        "TrackDecorator": {
+            "type": "object",
+            "properties": {
+                "type": { "const": "track" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "WireDecorator": {
+            "type": "object",
+            "properties": {
+                "type": { "const": "wire" },
+                "location": { "$ref": "#/definitions/SourceLocation" },
+                "adapterId": { "type": "string" },
+                "adapterModule": { "$ref": "#/definitions/ModuleReference" },
+                "adapterConfig": { "type": "object" }
+            }
+        }
+    }
+}
+```
+# Open questions
+* Should gathering metadata about configuration(`*.js-meta.xml`) file in a bundle be in the scope of this RFC?
```

Comment:
I don't think we should include the *.js-meta.xml yet:
1. Large parts of the metadata in *.js-meta.xml will be revamped based on the Canonical UI Metadata RFC here:  https://salesforce.quip.com/ZJ93A0XQZvnU. If you include the *.js-meta.xml now it'll be enshrining the old system.
2. Some of the information in it is pretty Salesforce-specific and my impression here is that you're going for an open source thing.

Maybe in the future, when this information is more based on a standard and less Salesforce specific.